lib/rules/no-restricted-imports.js

```diff
@@ -235,9 +242,10 @@ module.exports = {
 
         // relative paths are supported for this rule
         const restrictedPatternGroups = restrictedPatterns.map(
-            ({ group, message, caseSensitive, importNames, importNamePattern, allowImportNames, allowImportNamePattern }) => (
+            ({ group, regex, message, caseSensitive, importNames, importNamePattern, allowImportNames, allowImportNamePattern }) => (
                 {
                     matcher: ignore({ allowRelativePaths: true, ignorecase: !caseSensitive }).add(group),
```

Comment:
```suggestion
                    ...(group ? { matcher: ignore({ allowRelativePaths: true, ignorecase: !caseSensitive }).add(group) } : {}),
```

Since the `group` property is now not mandatory, it might make sense to create `matcher` conditionally, the same as with `regexMatcher`.